Document distributed tracing experiments (#199)
giantswarm/handbook: content/docs/observability/tracing/_index.md

For our initial experiment, we decided to work without any "tracing agent" like the opentelemetry-collector to reduce our footprint and effort but this might come in the future.

### On-demand

It can be enabled at the installation level like so https://github.com/giantswarm/config/blob/4e01cbc6de32420d111d78e0e278e72a31e5daba/installations/gaia/config.yaml.patch#L9

### Ingesting traces

#### Loki

If tracing is enabled at the installation level, Loki running on the MC is configured to send its traces to Tempo

### Prometheus

- To enable tracing for Prometheus, you should set the following configuration on the Prometheus CR:

```yaml
spec:
  tracingConfig:
    clientType: http
    endpoint: tempo-gateway.tempo.svc:80 # Tempo service on the MC
    insecure: false
    samplingFraction: "0.5" # sampling of 1/2 so we only send half of the traces
```

## Autoinstrumentation with eBPF

Distributed tracing is not a feature that comes by default in applications. They need to be instrumented by opentelemetry SDKs to be able to export traces.

> But there are cases where it is not possible to get traces from applications (blackbox or vendored applications, languages not equipped with opentelemetry SDKs like R). This is a really big blind spot in distributed tracing. The good news is that we can now rely on eBPF to get some basic traces from our applications. After experimenting with a few different tools like odigos, we tried out [Grafana Beyla](https://github.com/grafana/beyla) after in was released in version 1.0.0 and it looks to be the more mature solution (not relying on a lot of things that can fail and actually exporting traces from apps)

Reviewer: ```suggestion
But there are cases where it is not possible to get traces from applications (blackbox or vendored applications, languages not equipped with opentelemetry SDKs like R). This is a really big blind spot in distributed tracing. The good news is that we can now rely on eBPF to get some basic traces from our applications. After experimenting with a few different tools like odigos, we tried out [Grafana Beyla](https://github.com/grafana/beyla) after it was released in version 1.0.0 and it looks to be the most mature solution (not relying on a lot of things that can fail and actually exporting traces from apps)
```

> We ran it as an experiment for a customer but if the need arise, this is how we deployed it to trace an application written in R alongside an opentelemetry collector:

Reviewer: ```suggestion
We ran it as an experiment for a customer but if the need arises, the following shows how we deployed it to trace an application written in R alongside an opentelemetry collector:
```
